src/O1Heap.h

```diff
@@ -5,42 +5,44 @@
  * Contributors: https://github.com/107-systems/107-Arduino-Cyphal/graphs/contributors.
  */
 
-#ifndef ARDUINO_O1_HEAP_HPP_
-#define ARDUINO_O1_HEAP_HPP_
+#ifndef ARDUINO_O1_HEAP_H_
+#define ARDUINO_O1_HEAP_H_
 
 /**************************************************************************************
  * INCLUDE
  **************************************************************************************/
 
 #include <o1heap/o1heap.h>
 
+/**************************************************************************************
+ * DEFINE
+ **************************************************************************************/
+
+#define CYPHAL_DECLARE_HEAP(name,size) \
+  static uint8_t name[size] __attribute__ ((aligned (O1HEAP_ALIGNMENT)))
```

Comment:
In modern C++, there are few valid cases for the use of macros; all known to me modern coding standards strongly discourage their usage for anything except header inclusion and inclusion guards (even those are soon to go the way of the `goto` statement as C++20 finally has modules, yay).

```c++
template <std::size_t Size>
struct alignas(O1HEAP_ALIGNMENT) CyphalHeap final : public std::array<std::uint8_t, Size> {};
```

Reply:
With you on macros, C++20 is not an option as none of the currently supported Arduino cores are supporting C++20 (or even C++17/14 for that matter ... C++11 is a baseline we can take for granted). I'm testing your proposal in the CI build across all cores now.

Reply:
Derived from std::array works across all cores, see [here](https://github.com/107-systems/107-Arduino-Cyphal/actions/runs/3701421788/jobs/6270710791).